Neuro/Recording/LocalPlayer/LocalPlayerFrame.proto

```diff
@@ -13,4 +14,6 @@ message LocalPlayerFrame {
     repeated float RaycastObstacleDistances = 6;
     Vector2 Position = 7;
     Vector2 Velocity = 8;
+    bool IsDead = 9;
+    RoleType Role = 10;
```

Comment:
We are already recording this data when the game starts in HeaderFrame. All you need to do is update  it when the player dies. Creating a new frame type such as LocalPlayerDiedFrame would be a better solution imho.